File: backend/app/api/v1/reports.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.api.deps import get_current_user
from app.models.user import User
from app.orm.session import get_db
from app.schemas.params import DailySummaryParams, ReportRangeParams
from app.schemas.report import DailySummaryOut, MicrosReportOut, WeeklyReportOut
from app.services import report_service

router = APIRouter(prefix="/reports", tags=["reports"])


@router.get("/daily_summary", response_model=DailySummaryOut)
def daily_summary(
    params: DailySummaryParams = Depends(),
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    return report_service.get_daily_summary(db, current_user.id, params.date, params.tz)


@router.get("/weekly", response_model=WeeklyReportOut)
def weekly_report(
    params: ReportRangeParams = Depends(),
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    start_d, end_d, user_tz = report_service.resolve_range(params.week_of, params.start, params.end, params.tz)
    return report_service.get_weekly_report(db, current_user.id, start_d, end_d, user_tz)


@router.get("/micros", response_model=MicrosReportOut)
def micros_report(
    params: ReportRangeParams = Depends(),
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    start_d, end_d, user_tz = report_service.resolve_range(params.week_of, params.start, params.end, params.tz)
    return report_service.get_micros_report(db, current_user.id, start_d, end_d, user_tz)

File: frontend/src/api/reports.ts
import { useQuery } from '@tanstack/react-query'
import { USER_TZ } from '@/lib/tz'
import type { DailySummaryOut, MicrosReportOut, WeeklyReportOut } from '@/types/reports'
import { apiClient } from './client'

export function useDailySummary(date: string) {
  return useQuery({
    queryKey: ['daily-summary', date, USER_TZ],
    queryFn: () =>
      apiClient
        .get<DailySummaryOut>('/reports/daily_summary', { params: { date, tz: USER_TZ } })
        .then((r) => r.data),
  })
}

export function useWeeklyReport(weekOf: string) {
  return useQuery({
    queryKey: ['weekly-report', weekOf, USER_TZ],
    queryFn: () =>
      apiClient
        .get<WeeklyReportOut>('/reports/weekly', { params: { week_of: weekOf, tz: USER_TZ } })
        .then((r) => r.data),
  })
}

export function useWeeklyReportRange(start: string, end: string) {
  return useQuery({
    queryKey: ['weekly-report-range', start, end, USER_TZ],
    queryFn: () =>
      apiClient
        .get<WeeklyReportOut>('/reports/weekly', { params: { start, end, tz: USER_TZ } })
        .then((r) => r.data),
    enabled: !!start && !!end,
  })
}

export function useMicrosReport(start: string, end: string) {
  return useQuery({
    queryKey: ['micros-report', start, end, USER_TZ],
    queryFn: () =>
      apiClient
        .get<MicrosReportOut>('/reports/micros', { params: { start, end, tz: USER_TZ } })
        .then((r) => r.data),
    enabled: !!start && !!end,
  })
}
